packages/main/src/RPA/Robocorp/WorkItems.py

```diff
@@ -359,20 +366,27 @@ def output_path(self):
 
         return self._output_path
 
-    def create_output(self, parent_id: str, payload: Optional[JSONType] = None) -> str:
-        del parent_id
-
-        item: Dict[str, Any] = {"payload": payload, "files": {}}
-        self.outputs.append(item)
-
-        logging.debug("Payload: %s", json_dumps(payload, indent=4))
+    def _save_to_disk(self, source: str) -> None:
```

Comment:
Even easier to read code would be to have _save_input_to_disk / _save_output_to_disk and the do the file operation on the _save_to_disk function, would save a condition check, but that is matter of opinion of course :), anyway better this way than before. Just wanted to give any feedback :).

Reply:
Yes, I like this separation you suggest! Too bad I don't see how can I simplify the code within the current infrastructure. The culprit starts from this:
```python
    def _get_item(self, item_id: str) -> Tuple[str, Dict[str, Any]]:
        idx = int(item_id)
        if idx < len(self.inputs):
            return "input", self.inputs[idx]

        if idx < (len(self.inputs) + len(self.outputs)):
            return "output", self.outputs[idx - len(self.inputs)]

        raise ValueError(f"Unknown work item ID: {item_id}")
```
So solving that first might lead to a better codebase allowing this proposed refactor to happen.